app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -191,78 +163,67 @@ export const snapKeyringBuilder = (
                 'AccountsController:setSelectedAccount',
                 internalAccount.id,
               );
-              await controllerMessenger.call('ApprovalController:showSuccess', {
-                header: [snapAuthorshipHeader],
-                title: t('snapAccountCreated') as string,
-                icon: IconName.UserCircleAdd,
-                message: [
-                  {
-                    name: 'SnapAccountCard',
-                    key: 'snapAccountCard',
-                    properties: {
-                      address,
-                    },
-                  },
-                  {
-                    name: 'Text',
-                    key: 'description',
-                    children: [
-                      t('snapAccountCreatedDescription') as string,
-                      ' ',
-                      learnMoreLink,
-                    ],
-                  },
-                ],
-              });
-            } catch (error) {
-              const subjectMetadata = controllerMessenger.call(
-                'SubjectMetadataController:getSubjectMetadata',
-                origin,
+
+              // User should now see the "Successfuly added account" page
+              trackSnapAccountEvent(
```

Comment:
Note for the reviewers:

I thought of maybe augmenting the `showSuccess` function (or event deeper, to the component itself) to support those events, like having a:
```
events: {
  onLoad: MetaMetricsEventName.AddSnapAccountSuccessViewed,
  onSubmit: MetaMetricsEventName.AddSnapAccountSuccessClicked
},
```

So that our component would take care "hooking" those actions and fire events automatically. But this requires a bit more effort to implement properly.

I'm also open for suggestion here, since having those metrics in the `snap-keyring.ts` feels a bit wrong to me.

WDYT?

Reply:
I really like this idea and I agree that the implementation would scope creep this task a bit. If you strongly feel that it should be done first then I recommend opening a new PR for that change

Reply:
Since I'd like to move on with this PR, I'll probably keep it that way for now, and I'll leave a `TODO` to improve that later on.